api/Controllers/HotelsController.cs

```diff
@@ -0,0 +1,65 @@
+﻿using iTechArt.Hotels.Api.Models;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace iTechArt.Hotels.Api.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class HotelsController : Controller
+    {
+        private readonly HotelsDatabaseContext _hotelsDb;
+
+        public HotelsController(HotelsDatabaseContext hotelsDb)
+        {
+            _hotelsDb = hotelsDb;
+        }
+
+        [HttpPost]
+        [Authorize(Roles = "admin")]
+        public async Task<IActionResult> CreateHotel([FromBody] Hotel request )
+        {
+            if (request == null)
+            {
+                return BadRequest("Wrong response");
+            }
+
+            Hotel hotel = new Hotel
+            {
+                Name = request.Name,
+                Country = request.Country,
+                Sity = request.Sity,
+                Address = request.Address
+            };
+            _hotelsDb.Add(hotel);
+            await _hotelsDb.SaveChangesAsync();
+            return CreatedAtAction(nameof(GetHotel), new { id = hotel.HotelId }, null);
+        }
+
+        [Route("{id}")]
+        [HttpGet]
+        public async Task<IActionResult> GetHotel([FromRoute] int id)
+        {
+            Hotel hotel = await _hotelsDb.Hotels.SingleOrDefaultAsync(h => h.HotelId == id);
+            return Ok(hotel);
```

Comment:
What can go wrong using db entities as a return type? Didn't we already see that in authentication?
Also why are you using IActionResult as a return type? What are the alternatives?

Reply:
Not this method is used just for 204 code answer for post request. In future, if user clicks on hotel, he will get the whole hotel information. The reason for this is that I do not store any information that should not be available for clients, but I understand that such information might exist. So if you say that I should store something private, I will make one more hotel dto class

Reply:
I cannot see Dto class use, any updates?

Reply:
I have explained why I use not dto nut db model.
If next pr this method will be used to get whole hotel information

Reply:
Could you please let me know the reason too? Also I am not sure how next PR will change that/

Reply:
We have discussed this point with Alex, now I understand what you both talked about